lib/request_overrider.js

```diff
@@ -573,4 +555,40 @@ function RequestOverrider(req, options, interceptors, remove, cb) {
   return req
 }
 
+function parseResponseFromCallbackResult(response, callbackResult) {
+  debug('full response from callback result: %j', callbackResult)
+
+  if (!Array.isArray(callbackResult)) {
+    throw Error('A single function provided to .reply MUST return an array')
+  }
+
+  if (callbackResult.length > 3) {
+    throw Error(
+      'The array returned from the .reply callback contains too many values'
+    )
+  }
+
+  const [status, body, headers] = callbackResult
+
+  response.statusCode = common.statusCodeInt(status)
+
+  if (headers) {
+    debug('new headers: %j', headers)
+
+    const rawHeaders = Array.isArray(headers)
+      ? headers
+      : _.flatten(Object.entries(headers))
+    response.rawHeaders.push(...rawHeaders)
+
+    const castHeaders = Array.isArray(headers)
+      ? common.headersArrayToObject(headers)
+      : common.headersFieldNamesToLowerCase(headers)
+
+    _.assign(response.headers, castHeaders)
```

Comment:
Since this is being rewritten, could you replace it with a native version?

```js
response.headers = {
  ...response.headers,
  ...castHeaders
}
```